src/Cake.Common/Tools/NuGet/Pack/NuSpecTransformer.cs

```diff
@@ -61,29 +95,40 @@ private static XmlNode FindOrCreateElement(XmlDocument document, XmlNamespaceMan
                 if (parent == null)
                 {
                     // Get the package element.
-                    var package = document.SelectSingleNode("/package");
-                    if (package == null)
-                    {
-                        throw new CakeException("Nuspec file is missing package root.");
-                    }
+                    var package = GetPackageElement(document);
 
                     // Create the metadata element.
-                    parent = document.CreateElement("metadata", "http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd");
+                    parent = document.CreateElement("metadata", NuSpecXsd);
                     package.PrependChild(parent);
                 }
-
-                // If the parent didn't have a namespace specified, then skip adding one.
-                // Otherwise add the parent's namespace. This is a little hackish, but it 
-                // will avoid empty namespaces. This should probably be done better...
-                node = parent.NamespaceURI == string.Empty 
-                    ? document.CreateElement(name) 
-                    : document.CreateElement(name, parent.NamespaceURI);
-
-                parent.AppendChild(node);
+                
+                node = document.CreateAndAppendElement(parent, name);
             }
             return node;
         }
 
+        private static XmlNode CreateAndAppendElement(this XmlDocument document, XmlNode parent, string name)
+        {
+            // If the parent didn't have a namespace specified, then skip adding one.
+            // Otherwise add the parent's namespace. This is a little hackish, but it 
+            // will avoid empty namespaces. This should probably be done better...
+            return parent.AppendChild(
+                string.IsNullOrWhiteSpace(parent.NamespaceURI)
+                    ? document.CreateElement(name)
+                    : document.CreateElement(name, parent.NamespaceURI)
+                );
+        }
+
+        private static void AddAttributeIfSpecified(this XmlNode element, string value, string name)
+        {
+            if (string.IsNullOrWhiteSpace(value) || element.OwnerDocument==null || element.Attributes==null)
+                return;
```

Comment:
Actually, taking the time as I write this to finally add StyleCop. Thanks for the push, I needed it :wink: